7a_temp_coop_munge/src/data_parsers/parse_mndnr_files.R

```diff
@@ -44,3 +44,108 @@ parse_mndnr_files <- function(inind, outind) {
+  out <- readr::read_csv(file_path, col_types = 'cnccnccnccccncc') %>%
+    dplyr::filter(Parameter == 'Temperature, water') %>%
+    dplyr::select(Lake_Name, DOWLKNUM, Date, Time,
+                  Result, Upper_Depth) %>%
+    dplyr::rename(DateTime = Date,
+                  time = Time,
+                  depth = Upper_Depth,
+                  temp = Result,
+                  DOW = DOWLKNUM,
+                  site = Lake_Name
+    ) %>%
+    mutate(DateTime = as.Date(strptime(DateTime, "%m/%d/%Y")),
+           time = substr(as.character(time), 1, 5))
+
+  return(out)
+}
+
+
+# parser and helper functions for Long29016100.zip -----------
+parse_Long_29016100 <- function(inind, outind) {
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  # Unzip the file with all of Bull Shoals, then cleanup unzipped files
+  # which aren't needed externally before leaving the function.
+  unzip_dir <- tempdir()
+  files_from_zip <- unzip(infile, exdir = unzip_dir)
+  on.exit(unlink(unzip_dir, recursive = TRUE))
+
+
+  clean <- files_from_zip %>%
+    purrr::map_dfr(~ clean_Long29016100(file_path = .x,
+                                        skip = 2, date_cell = 'E1',
+                                        omit_sheets = c('Summary', 'Depths',
+                                                        'Layers', '8-12-16 (Sta 1 So.)',
+                                                        '8-12-16 (Sta 3 No.)')))
+
+  saveRDS(object = clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+}
+
+#' @param file_path chr, full file path for file
+#' @param skip int, number of rows to skip
+#' @param date_cell chr, Excel-style cell ID for the sample date
+#' @param omit_sheets chr, vector of sheets that should be omitted if encountered
+#'
+clean_Long_29016100 <- function(file_path, skip = 2,
+                               date_cell = 'E1', omit_sheets = NULL) {
+
+  # read sheets and remove omitted sheets
+  sheets <- readxl::excel_sheets(file_path) %>%
+    .[!(. %in% omit_sheets)]
+
+  clean <- sheets %>%
+    map_dfr(~ build_sheet_data(file_path = file_path, sheet = .x,
+                               date_cell = date_cell, skip = skip))
+}
+
+#' @param file_path chr, full file path for file
+#' @param sheet chr, name of the sheet that should be read in
+#' @param skip int, number of rows to skip
+#' @param date_cell chr, Excel-style cell ID for the sample date
+#'
+build_sheet_data <- function(file_path, sheet, date_cell, skip) {
+  # extract sample date
+  sample_date <- readxl::read_excel(path = file_path,
+                                    sheet = sheet,
+                                    range = date_cell,
+                                    col_names = F) %>% dplyr::pull()
+
+  # read sheet data, tack on date and DOW
+  dat <- readxl::read_excel(path = file_path, sheet = sheet, skip = skip) %>%
+    dplyr::select(matches('depth m|oC'))
```

Comment:
If the name of the depth column varies, would this `matches('depth m')` work as you expect? For example, the following will only return one column. I wonder if you should use `contains()` instead.

```r
tibble(`depth` = 1:4, 
       `depth (m)` = 1:4, 
       `depth m` = 1:4) %>% 
    select(matches('depth m'))

# A tibble: 4 x 1
  `depth m`
      <int>
1         1
2         2
3         3
4         4
```

vs with `contains()`

```r
tibble(`depth` = 1:4, 
       `depth (m)` = 1:4, 
       `depth m` = 1:4) %>% 
    select(contains('depth'))

# A tibble: 4 x 3
  depth `depth (m)` `depth m`
  <int>       <int>     <int>
1     1           1         1
2     2           2         2
3     3           3         3
4     4           4         4
```

Reply:
Interestingly, `contains` and `matches` return the same results:

```r
t1 <- tibble(`depth` = 1:4, 
       `depth (m)` = 1:4, 
       `depth m` = 1:4) %>% 
  select(matches('depth m'))
  
t2 <- tibble(`depth` = 1:4, 
             `depth (m)` = 1:4, 
             `depth m` = 1:4) %>% 
  select(contains('depth m'))

all.equal(t1, t2)

> all.equal(t1, t2)
[1] TRUE
```
Based on the doc for the two functions, they can both do partial matches. It looks like the key difference is that `contains` matches only strings while `matches` has the ability to use regular expressions. Here is what that comparison looks like when I use a regular expression to grab two columns (depth and temp):

```r
tibble(`depth` = 1:4, 
             `depth (m)` = 1:4, 
             `depth m` = 1:4,
             `oC` = 1:4) %>% 
  select(matches('depth|oC'))

## RESULT
# A tibble: 4 x 4
  depth `depth (m)` `depth m`    oC
  <int>       <int>     <int> <int>
1     1           1         1     1
2     2           2         2     2
3     3           3         3     3
4     4           4         4     4

tibble(`depth` = 1:4, 
             `depth (m)` = 1:4, 
             `depth m` = 1:4,
             `oC` = 1:4) %>%
  select(contains('depth|oC'))

## RESULT 
# A tibble: 4 x 0

```

With this specific set of files I have to keep the `m` in `depth m` because this data reports metric and imperial units in the same spreadsheets (e.g., `Depth m` and `Depth ft`).

Interestingly, while answering this question, i found that `matches` and `contains` are not case sensitive:
```r
> tibble(`depth` = 1:2, 
+              `depth (m)` = 1:2, 
+              `depth m` = 1:2,
+              `Depth m` = 1:2,
+              `oC` = 1:2) %>% 
+   select(matches('depth|oC'))
# A tibble: 2 x 5
  depth `depth (m)` `depth m` `Depth m`    oC
  <int>       <int>     <int>     <int> <int>
1     1           1         1         1     1
2     2           2         2         2     2


> tibble(`depth` = 1:2, 
+              `depth (m)` = 1:2, 
+              `depth m` = 1:2,
+              `Depth m` = 1:2,
+              `oC` = 1:2) %>% 
+   select(contains('depth'))
# A tibble: 2 x 4
  depth `depth (m)` `depth m` `Depth m`
  <int>       <int>     <int>     <int>
1     1           1         1         1
2     2           2         2         2

```

This is due to an internal call to `tolower` (for `contains`) and `ignore.case` (for matches):
```r
> contains
function (match, ignore.case = TRUE, vars = NULL) 
{
    check_match(match)
    vars <- vars %||% peek_vars(fn = "contains")
    if (ignore.case) {
        vars <- tolower(vars)
        match <- tolower(match)
    }
    flat_map_int(match, grep_vars, vars, fixed = TRUE)
}
<bytecode: 0x000001e7fdfb5988>
<environment: namespace:tidyselect>

> matches
function (match, ignore.case = TRUE, perl = FALSE, vars = NULL) 
{
    check_match(match)
    vars <- vars %||% peek_vars(fn = "matches")
    flat_map_int(match, grep_vars, vars, ignore.case = ignore.case, 
        perl = perl)
}
<bytecode: 0x000001e7fdf8e8a0>
<environment: namespace:tidyselect>
```

Reply:
I wasn't sure how your columns varied, but was concerned about the potential for a `depth (m)` vs a `depth m` and using `matches('depth m')` would drop `depth (m)`. Sounds like that's not the case and we are good to go! Wanted to double check and looks like you've quadruple checked here 😄